v2/src/lib.rs

```diff
@@ -0,0 +1,837 @@
+        Ok(_) => Ok(true),
+        Err(_) => Ok(false), //can post error here
+    }
+}
+// Returns an InfluxDB regex (see https://docs.influxdata.com/influxdb/v1.8/query_language/explore-data/#regular-expressions)
+// corresponding to the list of path expressions. I.e.:
+// Replace "**" with ".*", "*" with "[^\/]*"  and "/" with "\/".
+// Concat each with "|", and surround the result with '/^' and '$/'.
+fn key_exprs_to_influx_regex(path_exprs: &[&keyexpr]) -> String {
+    let mut result = String::with_capacity(2 * path_exprs[0].len());
+    result.push_str("/^");
+    for (i, path_expr) in path_exprs.iter().enumerate() {
+        if i != 0 {
+            result.push('|');
+        }
+        let mut chars = path_expr.chars().peekable();
+        while let Some(c) = chars.next() {
+            match c {
+                '*' => {
+                    if let Some(c2) = chars.peek() {
+                        if c2 == &'*' {
+                            result.push_str(".*");
+                            chars.next();
+                        } else {
+                            result.push_str(".*")
+                        }
+                    }
+                }
+                '/' => result.push_str(r"\/"),
+                _ => result.push(c),
+            }
+        }
+    }
+    result.push_str("$/");
+    result
+}
+
+fn time_from_parameters(t: &str) -> ZResult<Option<(u64, u64)>> {
+    use zenoh::selector::{TimeBound, TimeRange};
+    let time_range = t.time_range()?;
+
+    let mut start_time: u64 = 0;
+    let mut stop_time: u64 = Utc::now().naive_utc().timestamp().try_into().unwrap();
+
+    match time_range {
+        Some(TimeRange(start, stop)) => {
+            match start {
+                TimeBound::Inclusive(t) => {
+                    start_time = calculate_time(t).unwrap();
+                }
+                TimeBound::Exclusive(t) => {
+                    start_time = calculate_time(t).unwrap();
+                }
+                TimeBound::Unbounded => {}
+            }
+            match stop {
+                TimeBound::Inclusive(t) => {
+                    stop_time = calculate_time(t).unwrap();
+                }
+                TimeBound::Exclusive(t) => {
+                    stop_time = calculate_time(t).unwrap();
+                }
+                TimeBound::Unbounded => {}
+            }
+        }
+        None => {
+            return Ok(None);
+        }
+    }
+    Ok(Some((start_time, stop_time)))
+}
+
+fn calculate_time(tx: TimeExpr) -> ZResult<u64> {
+    //Note: conversions bw f64,i64,u64
+    let now_time = Utc::now().naive_utc().timestamp();
+    match tx {
+        TimeExpr::Fixed(t) => {
+            let time_in_secs = t
+                .duration_since(UNIX_EPOCH)
+                .expect("Time went backwards")
+                .as_secs();
+            Ok(time_in_secs)
+        }
+        TimeExpr::Now { offset_secs } => Ok((now_time + offset_secs as i64).try_into().unwrap()),
```

Comment:
Unwrapping while a Result is returned doesn't make sense. The underlying error can be returned, preferably with an explicit message.

Reply:
Returned underlying error.